src/ui/components/RecommendedBadge/index.js

```diff
@@ -1,25 +1,41 @@
 /* @flow */
+import makeClassName from 'classnames';
 import * as React from 'react';
 import { compose } from 'redux';
 
 import translate from 'core/i18n/translate';
-import Icon from 'ui/components/Icon';
+import IconRecommendedBadge from 'ui/components/IconRecommendedBadge';
+import type { RecommendedBadgeSize } from 'ui/components/IconRecommendedBadge';
 import type { I18nType } from 'core/types/i18n';
 
 import './styles.scss';
 
-type Props = {};
+type Props = {|
+  size: RecommendedBadgeSize,
+|};
 
 type InternalProps = {|
+  ...Props,
   i18n: I18nType,
 |};
 
-export const RecommendedBadgeBase = (props: InternalProps) => {
-  const { i18n } = props;
-
+export const RecommendedBadgeBase = ({ i18n, size }: InternalProps) => {
   const label = i18n.gettext('Recommended');
+
+  let sizeClassName;
```

Comment:
See my suggestion above about replacing this `switch` with something more concise. You could probably even move the logic for creating the `sizeClassName` into a function in `IconRecommendedBadge` which you could then call from here, replacing all of this code with a single line.

Reply:
The class names are different in each place so I don't think a shared function would help much.